woocommerce/woocommerce-gutenberg-products-block
# Shopper → Checkout → Can apply single-use coupon once

Fixes #5773

### Notes

This PR adds an e2e test to verify that a "single-use" coupon can be used once.

### Testing

#### Automated Tests

### Manual Testing

If the E2E tests run by GitHub actions are successful in this PR then it's all good.

As an extra, if you want to test it locally as well:

1. Check out this PR.
2. Start Docker Desktop.
3. Run `npm run wp-env destroy && npm run wp-env start` to start a fresh e2e environment.
4. Run `npm run test:e2e -- tests/e2e/specs/shopper/checkout-single-use-coupon.test.js` to run only this test case.
5. Run `npm run test:e2e` to run the whole test suite.

### Rest API instead of Fixture

We are creating our "single-use" coupon using the Woo's Rest API instead of including it in the fixture data (`tests/e2e/fixtures/fixture-data.js`) to avoid using it in other tests.
